Normalize to Unicode NFC encoding before converting accent characters

ironprogrammer opened the pull request:
Refresh of [previous patch for #24661](https://core.trac.wordpress.org/attachment/ticket/24661/normalize_before_removing_accents.patch), with addition of unit test to convert NFD-encoded strings.

Trac ticket: https://core.trac.wordpress.org/ticket/24661

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

ironprogrammer committed:
Refresh for Trac #24661

Position of added snippet moved.

ironprogrammer committed:
Add `@since` comment

ironprogrammer committed:
Add test for NFD-encoded Latin1 Supplement chars

Note: these are NOT the same encodings used for the original `test_remove_accents_latin1_supplement()` function, though they are visually identical.

azaozz commented on tests/phpunit/tests/formatting/removeAccents.php:
```diff
@@ -109,4 +109,20 @@ public function test_transcribe_serbian_crossed_d() {
+		$input  = 'ªºÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ';
+		$output = 'aoAAAAAAAECEEEEIIIIDNOOOOOOUUUUYTHsaaaaaaaeceeeeiiiidnoooooouuuuythy';
+
+		$this->assertSame( $output, remove_accents( $input ), 'remove_accents replaces Latin-1 Supplement with NFD encoding' );
```
Looks good! Just a nitpick :) 
If the functionality depends on `function_exists()` (`intl` support in PHP), perhaps the test should too? Can probably be marked as skipped if no support.

ironprogrammer replied:
Good tip 👍🏻 Thanks!

I've dropped in a [`@requires`](https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#skipping-tests-using-requires) to make sure `intl` is supported (c78aedb), else it will skip the test.

ironprogrammer committed:
Add `@requires` annotation for `intl` support

Test is skipped if required extension isn't present.

Only impacts non-NFC encoded tests, like this one.

ironprogrammer committed:
Add `@covers` for Tests_Formatting_RemoveAccents

azaozz commented on src/wp-includes/formatting.php:
```diff
@@ -1597,6 +1598,11 @@ function remove_accents( $string, $locale = '' ) {
 	}
 
 	if ( seems_utf8( $string ) ) {
```
Thinking an inline comment would probably be useful here too. Perhaps something like (copied from the tests):

```
// Unicode sequence normalization from NFD (Normalization Form Decomposed)
// to NFC (Normalization Form [Pre]Composed), the encoding used in this function.
```

ironprogrammer replied:
Good suggestion 👍🏻 Updated!

ironprogrammer committed:
Add inline comment for introduced NFD to NFC normalization

audrasjb commented:
Closed in favor of https://github.com/WordPress/wordpress-develop/pull/3000

audrasjb commented:
Committed in https://core.trac.wordpress.org/changeset/53754
